addons/packages/kapp-controller/0.30.0/bundle/config/overlays/update-deployment.yaml

```diff
@@ -29,15 +29,42 @@ spec:
           #@overlay/match by="name"
           - name: KAPPCTRL_API_PORT
             value: #@ str(values.kappController.deployment.apiPort)
+        #@ if values.kappController.deployment.hostNetwork:
+        volumeMounts:
+          - mountPath: /etc
+            name: etc
+        #@ end
       #@ if values.kappController.deployment.hostNetwork:
-      hostNetwork: #@ values.kappController.deployment.hostNetwork
-      dnsPolicy: ClusterFirstWithHostNet
+      #! Using init container bypasses the restriction of not having root access in main container
+      #! It modifies /etc/resolv.conf which is shared to main container
+      initContainers:
```

Comment:
Same comment as above add initContainers if coreDNSIP is specified.

Reply:
Changed